runtime-infra/lambdas/cdc-athena-view-cf-transform/src/app/hive-type-parser/BaseHiveTypeVisitor.js

```diff
@@ -0,0 +1,49 @@
+const { HiveType } = require("./HiveType.js");
+
+class BaseHiveTypeVisitor {
+
+  constructor() { }
+
+  visit( rootHiveType, context ) {
+    const nodes = []
+    if( rootHiveType ) {
+      nodes.unshift( rootHiveType )
+    }
+    else {
+      throw new Error("rootHiveType parameter is required")
+    }
+    
+    while( nodes.length > 0 ) {
+      const ht = nodes.shift();
+      const haveToDescend = this.enterNodeVisit( ht, context );
+      if ( haveToDescend ) {
+        if ( "STRUCT" == ht.category ) {
+          for( let fieldName of Object.keys( ht.structChildren ).sort().reverse() ) {
+            let fieldType = ht.structChildren[ fieldName ]
+            nodes.unshift( fieldType )
+          }
+        }
+        else if ( "ARRAY" == ht.category ) {
```

Comment:
the same

Reply:
Penso di averli corretti, in compenso avrò aggiunto un sacco di errori grammaticali nei README.md